.github/actions/validate-access/action.yaml

```diff
@@ -0,0 +1,53 @@
+name: Validate Access
+description: 'Check if the workflow is triggered by an admin user'
+
+# This callable workflow checks if the workflow is triggered by
+# a code owner or an image maintainer by testing the github.actor
+# variable against the CODEOWNERS file and the contacts.yaml file
+# under oci/* path
+
+inputs:
+  admin-only:
+    description: 'The protected workflow should only be triggered as a code owner or an image maintainer'
+    required: true
+    default: 'false'
+  image-path:
+    description: 'The path to the image to be built'
+    required: true
+  github-token:
+    description: 'The GH_TOKEN for the GitHub CLI'
+    required: true
+
+runs:
+  using: "composite" 
+  steps:
+    - name: Check if the workflow is triggered by an admin user
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.github-token }}
+      run: |
+        echo "github.actor: ${{ github.actor }}"
+        echo "admin-only: ${{ inputs.admin-only }}"
+        if [[ ${{ inputs.admin-only }} == true ]]; then
+          echo "Expanding team mentions in the CODEOWNERS file"
+          teams=$(grep -oE '@[[:alnum:]_.-]+\/[[:alnum:]_.-]+' ${{ github.workspace }}/CODEOWNERS || true | sort | uniq)
+
+          for team in $teams; do
+            org=$(echo $team | cut -d'/' -f1 | sed 's/@//')
+            team_name=$(echo $team | cut -d'/' -f2)
```

Comment:
I'm not sure about the name restrictions on $team, but maybe we should double quote it to be safe?

Reply:
IMO, we don't have to, since the $team does not contain white spaces.